service/api.go

```diff
@@ -322,7 +361,8 @@ func (as *APIServer) GetServiceApp() *gimlet.APIApp {
 	app.SimpleVersions = true
 
 	// Project lookup and validation routes
-	app.AddRoute("/ref/{identifier}").Wrap(requireUser).Handler(as.fetchProjectRef).Get()
+	app.AddRoute("/ref/{projectId}").Wrap(requireUser).Handler(as.fetchLimitedProjectRef).Get()
```

Comment:
Can we add a deprecation notice to this to warn devs not to extend our usage of this route any more than it's already used?

Reply:
What do you mean by that? This should only be used by our cli, I don't think it's documented anywhere.

Reply:
It's true that they're only used by the CLI, but most of the CLI has already switched over to using `setupRestCommunicator` to get a client for REST v2 to facilitate our move away from REST v1. The REST v1 API has been deprecated for years in favor of REST v2 (i.e. the routes in `/rest/route/service.go`), and the only reason we keep it around is due to lack of benefit from migrating existing routes to v2. However, since we're writing new routes here, it seems like they should follow the REST v2 style.

Reply:
Hmm, this is true, but I still think we should leave a deprecation notice. Since this still bypasses the permissions system (in a more limited way), warning devs that `/ref/{project_id}` is solely for the sake preserving existing functionality working seems alright, but we should still discourage devs from trying to extend the usage of legacy routes for future needs (e.g. like stuffing more CLI functionality into the route).

Reply:
added one up top

Reply:
I added it to line 331

Reply:
Thanks!